src/Utils/MimeActions.vala

```diff
@@ -171,23 +167,19 @@ public class Marlin.MimeActions {
             GLib.List<AppInfo> one_result = get_applications_for_file (file);
             one_result.sort (application_compare_by_id);
 
-            if (result != null) {
+            if (result.length () > 0) {
```

Comment:
length iterates the whole list when you can do:
```suggestion
            if (result.data != null) {
```

Reply:
In this case the length is only a handful of apps so I wasn't concerned but I'll make these changes.